app/models/user.rb

```diff
@@ -192,6 +192,21 @@ def update_tags!
     Tag.where(user_id: id).update_all(user_id: nil, username: username)
   end
 
+  # Update the activities owned by this user. This method should only be called
+  # before destroying this user.
+  def update_activities!(owner)
+    PublicActivity::Activity.where(owner_id: id).find_each do |a|
+      a.owner_id   = nil
+      a.owner_type = nil
+      a.parameters = a.parameters.merge(owner_name: display_username)
+      a.save
+    end
```

Comment:
That's what we had before, but we need to:

```ruby
a.parameters = a.parameters.merge(owner_name: display_username)
```

That's one of the bugs that I caught 😉 Maybe a transaction would've been better so at least everything is done atomically.